test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java

```diff
@@ -142,7 +142,7 @@ public void afterEach(ExtensionContext context) throws Exception {
 
     private boolean isNativeTest(ExtensionContext context) {
         return context.getRequiredTestClass().isAnnotationPresent(NativeImageTest.class)
-                | context.getRequiredTestClass().isAnnotationPresent(SubstrateTest.class);
```

Comment:
I have removed this condition when I deleted the `SubstrateTest` class in https://github.com/quarkusio/quarkus/pull/5596